gflownet/envs/cube.py

```diff
@@ -18,70 +18,69 @@
+    If the value of a dimension gets larger than 1 - min_incr, then the trajectory is
+    ended (the only possible action is EOS).
 
     Attributes
     ----------
     n_dim : int
         Dimensionality of the hyper-cube.
 
-    max_val : float
-        Max length of the hyper-cube.
-
     min_incr : float
-        Minimum increment in the actions, expressed as the fraction of max_val. This is
-        necessary to ensure coverage of the state space.
+        Minimum increment in the actions, in (0, 1). This is necessary to ensure
+        that all trajectories have finite length.
+
+    n_comp : int
+        Number of components in the mixture of Beta distributions.
     """
 
     def __init__(
         self,
         n_dim: int = 2,
-        max_val: float = 1.0,
         min_incr: float = 0.1,
         n_comp: int = 1,
-        beta_params_min: float = 0.1,
-        beta_params_max: float = 1000.0,
         fixed_distr_params: dict = {
+            "beta_params_min": 0.1,
+            "beta_params_max": 1000.0,
             "beta_weights": 1.0,
             "beta_alpha": 2.0,
             "beta_beta": 5.0,
-            "bernoulli_source_logit": 1.0,
-            "bernoulli_eos_logit": 1.0,
+            "bernoulli_bts_prob": 1.0,
+            "bernoulli_eos_prob": 1.0,
         },
         random_distr_params: dict = {
+            "beta_params_min": 0.1,
+            "beta_params_max": 1000.0,
             "beta_weights": 1.0,
             "beta_alpha": 1000.0,
             "beta_beta": 1000.0,
-            "bernoulli_source_logit": 1.0,
-            "bernoulli_eos_logit": 1.0,
+            "bernoulli_bts_prob": 1.0,
+            "bernoulli_eos_prob": 1.0,
         },
         **kwargs,
     ):
         assert n_dim > 0
-        assert max_val > 0.0
+        assert min_incr > 0.0
+        assert min_incr < 1.0
         assert n_comp > 0
         # Main properties
         self.n_dim = n_dim
-        self.eos = self.n_dim
-        self.max_val = max_val
-        self.min_incr = min_incr * self.max_val
+        self.min_incr = min_incr
         # Parameters of the policy distribution
         self.n_comp = n_comp
-        self.beta_params_min = beta_params_min
-        self.beta_params_max = beta_params_max
-        # Source state: position 0 at all dimensions
-        self.source = [0.0 for _ in range(self.n_dim)]
-        # Action from source: (n_dim, 0)
-        self.action_source = (self.n_dim, 0)
-        # End-of-sequence action: (n_dim + 1, 0)
-        self.eos = (self.n_dim + 1, 0)
+        self.beta_params_min = fixed_distr_params["beta_params_min"]
+        self.beta_params_max = fixed_distr_params["beta_params_max"]
+        # Source state is abstract - not included in the cube: -1 for all dimensions.
+        self.source = [-1 for _ in range(self.n_dim)]
+        # Small constant to clamp the inputs to the beta distribution
+        self.epsilon = 1e-6
```

Comment:
I would do this as an argument, with one practical reason being, I would actually be quite interested in seeing the impact of this on the performance (I see you tested too values in the wandb report, but perhaps once everything else is finalized).

Reply:
I hate saying this, but I expected this comment haha ^^

There you go: 
- https://github.com/alexhernandezgarcia/gflownet/pull/214/commits/50046323f60a164b290e30a11d90abedea23e8b3
- https://github.com/alexhernandezgarcia/gflownet/pull/214/commits/117be0388e0ca1d5638205e64682480c328f228d

Note that there are two "epsilons" (epsilon and kappa). One is for clamping the increments that are inputs to the Beta distributions; one is for the limits of the test data sets, so as to reduce the influence of the loss of accuracy in fitting the regions close to the boundaries.